controllers/discovery/initDiscoveryUserData.js

```diff
@@ -72,6 +85,7 @@ const InitDiscoveryUserData = async (req, res) => {
       type: QueryTypes.SELECT,
       replacements: {
         userId,
+        topicIds,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Here, `topicIds` is passed as a replacement in the query. This is fine if `topicIds` is a single value. But if `topicIds` is an array (which seems to be the case), this might not work as expected. You need to convert the array into a format that can be used in the SQL query. For example, you could join the array elements into a comma-separated string.
```diff
-        topicIds,
+        topicIds: topicIds.join(','),
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->